docs/search-v2-change-indices.md

```diff
@@ -0,0 +1,128 @@
+2. Checkout your changes, then run `./manage.py es7_init`
+3. If this fails for a Document, then we know we have to create a new index for that Document type
+
+However,
+if it succeeds for a Document,
+this doesn't necessarily mean that it's safe to append the config change to that document in prod.
+This is because the mapping in prod could contain fields which have now been removed in the Document class.
+To be certain a mapping change can be appended without the need for a new index:
+
+1. [Get the mapping of the index in prod](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-get-mapping.html)
+2. [Get the settings of the index in prod](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-get-settings.html)
+3. [Create a new index locally](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html),
+using parts of the above responses:
+
+    ```
+    PUT /my-index-000001
+    {
+      "mappings": {
+        < prod_index_name/_mapping["prod_index_name"]["mappings"] >
+      },
+      "settings: {
+        "analyzers": {
+          < prod_index_name/_settings["prod_index_name"]["settings"]["index"]["analysis"] >
+        }
+      }
+    }
+    ```
+
+4. Point your Document at this index and run `./manage.py es7_init --limit DocumentName`
+
+If this succeeds,
+you can be relatively confident that this is a config change which doesn't require creating a new index.
+
+## Changes appended to a current index
+
+Your approach here will completely depend on what kind of changes you're making on the index.
+
+For example,
+if you're adding a new field which hasn't existed before,
+you'll want to make sure any changes you make which read or query ElasticSearch don't assume that the field is immediately available.
+This is because,
+until you've fully reindexed the documents in this index,
+documents will exist without the field you're adding.
+
+So if you were adding a `searchme` field to a document which defaults to true,
+rather than adding a filter to only include documents where `searchme=True`,
+you would want your query to exclude documents where `searchme=False`,
+in order to avoid having a moment during your deploy where nothing appears in the query results.
+
+```eval_rst
+.. Note::
+  If you're removing a field from a Document,
+  it would be preferable to leave it in the Document class,
+  setting its value on all documents to `None`,
+  with a note to remove it the next time a change requiring a new index is made.
+
+  This is to avoid the index on prod containing fields which don't exist on indices created afresh locally.
+```
+
+## Changes requiring a new index
+
+### Development process
+
+#### Config changes
+
+Make any changes to Documents, analyzers or index configuration which you need to do as usual,
+and increment the `Document.Index.version` attribute.
+
+#### Write operations
```

Comment:
Write operations should also use aliases and update both new and old indices. For the wiki index we could have the following:
* `new_wiki_index_alias -> wiki_index_timestamp_now`
* `current_wiki_index_alias  -> wiki_index_timestamp_old`

We always read and write from/to the `current_wiki_index_alias`.
When there is a write operation and the `new` index is defined we write there too.
When the reindex is complete we point current to `wiki_index_timestamp_new` and delete the reference to the new one.

Reply:
We can do this regardless if we are using timestamps or versions. My goal here is twofold: minimize user interaction in the process as this can lead to errors and also provide both fresh data to the old index or avoid stale data to the new one.